src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -697,6 +712,74 @@ std::string SQLiteDBEngine::buildLeftOnlyQuery(const std::string& t1,
     return std::string("SELECT "+fieldsList+" FROM "+t1+" t1 LEFT JOIN "+t2+" t2 ON "+onMatchList+" WHERE "+nullFilterList+";");
 } 
 
+void SQLiteDBEngine::getRowDiff(const std::string& table,
+                                const nlohmann::json& data,
+                                nlohmann::json& jsResult)
+{
+    std::vector<std::string> primaryKeyList;
+    if (getPrimaryKeysFromTable(table, primaryKeyList))
+    {
+        const auto& stmt
+        {
+            getStatement(buildSelectMatchingPKsSqlQuery(table, primaryKeyList))
+        };
+
+        const auto& tableFields { m_tableFields[table] };
+        int32_t index { 1l };
+        for (const auto& pkValue : primaryKeyList)
+        {
+            const auto& it
+            {
+                std::find_if(tableFields.begin(), tableFields.end(),
+                                [&pkValue](const ColumnData& column)
+                                {
+                                    return 0 == std::get<Name>(column).compare(pkValue);
+                                })
+            };
+
+            if(it != tableFields.end())
+            {
+                jsResult.push_back({{pkValue, data[pkValue]}});
```

Comment:
if the primary key exist and the call to step returns SQLITE_ROW but there is no difference between the stored data and the current data returned jsResult won't be empty triggering a wrong modified event. A flag can be set if some field has changed to not clear the jsResult before returning.

Reply:
Will do.